railties/test/application/test_runner_test.rb

```diff
@@ -249,11 +249,65 @@ def test_generated_scaffold_works_with_rails_test
       assert_match "0 failures, 0 errors, 0 skips", run_test_command('')
     end
 
+    def test_run_test_task_before_others
+      create_test_file :models, 'foo'
+      app_file 'lib/tasks/dummy.rake', <<-RUBY
+        task :dummy do
+          puts 'Dummy'
+        end
+      RUBY
+      run_tasks('test', 'dummy').tap do |output|
+        assert_match /FooTest.+Dummy/m, output
+        assert_match "1 runs, 1 assertions, 0 failures", output
+      end
+    end
+
+    def test_run_multiple_test_tasks_amongst_others
+      create_test_file :models, 'foo'
+      create_test_file :controllers, 'bar'
+      app_file 'lib/tasks/dummy.rake', <<-RUBY
+        task :dummy do
+          puts 'Dummy'
+        end
+      RUBY
+      run_tasks('test:models', 'dummy', 'test:controllers').tap do |output|
+        assert_match /FooTest.+Dummy.+BarTest/m, output
+        assert_match /(1 runs, 1 assertions, 0 failures.+){2}/m, output
+      end
+    end
+
+    def test_run_failing_test_task_omits_others
+      app_file "test/models/foo_test.rb", <<-RUBY
+        require 'test_helper'
+
+        class FooTest < ActiveSupport::TestCase
+          def test_falsity
+            puts "FooTest"
+            assert false
+          end
+        end
+      RUBY
+      app_file 'lib/tasks/dummy.rake', <<-RUBY
+        task :dummy do
+          puts 'Dummy'
+        end
+      RUBY
+      run_tasks('test', 'dummy').tap do |output|
+        assert_match "FooTest", output
+        assert_no_match "Dummy", output
+        assert_match "1 runs, 1 assertions, 1 failures", output
```

Comment:
As far as I can tell, we are only matching if those output exist, and not the order of them.
We should match the order too, to test that `at_exit` from Minitest is not running.